app/main.py

```diff
@@ -1,2 +1,100 @@
+from typing import Any, Generator
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self.max_capacity = 8
+        self.threshold = self.max_capacity - (self.max_capacity * 2 // 3)
+        self.hash_table = [None] * self.max_capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.hash_table.count(None) <= self.threshold:
```

Comment:
The condition `self.hash_table.count(None) <= self.threshold` might not correctly trigger the resize operation when needed. Consider revising this logic to ensure the table resizes appropriately when the load factor exceeds a certain threshold.